src/extension.ts

```diff
@@ -107,18 +107,46 @@ export function activate(context: ExtensionContext) {
     return await startServer(binaryPath);
   }
 
+  let workspaceFolder: vscode.WorkspaceFolder;
+
+  let rootPath: string = vscode.workspace.getConfiguration("smithyLsp").get("rootPath");
+
+  if (rootPath) {
+    const workspaceRoot = getWorkspaceRoot();
+    if (rootPath.startsWith("${workspaceRoot}") && workspaceRoot === "") {
+      console.warn(`Unable to retrieve the workspace root.`);
+    }
+    workspaceFolder = {
+      uri: vscode.Uri.file(rootPath.replace("${workspaceRoot}", workspaceRoot)),
+      name: "smithy-lsp-root-path",
+      index: 1,
+    };
+  }
+
+  // Configure file patterns relative to the workspace folder.
+  let filePattern: vscode.GlobPattern = "**/{smithy-build}.json";
+  let selectorPattern: string = null;
+  if (workspaceFolder) {
+    filePattern = new vscode.RelativePattern(workspaceFolder, filePattern);
+    selectorPattern = `${workspaceFolder.uri.fsPath}/**/*`;
+  }
+
   // Options to control the language client
   let clientOptions: LanguageClientOptions = {
     // Register the server for plain text documents
     documentSelector: [
-      { scheme: "file", language: "smithy" },
-      { scheme: "smithyjar", language: "smithy" },
+      { scheme: "file", language: "smithy", pattern: selectorPattern },
+      { scheme: "smithyjar", language: "smithy", pattern: selectorPattern },
     ],
     synchronize: {
       // Notify the server about file changes to 'smithy-build.json' files contained in the workspace
-      fileEvents: workspace.createFileSystemWatcher("**/{smithy-build}.json"),
+      fileEvents: vscode.workspace.createFileSystemWatcher(filePattern),
     },
     outputChannelName: "Smithy Language Server",
+    workspaceFolder,
+
+    // Don't switch to output window when the server returns output.
+    revealOutputChannelOn: RevealOutputChannelOn.Never,
   };
```

Comment:
This can all probably be moved to its own top-level function, something like `getClientOptions`, with a return type of `LanguageClientOptions`